src/main/java/mcscheduler/logic/parser/RoleEditCommandParser.java

```diff
@@ -0,0 +1,35 @@
+package mcscheduler.logic.parser;
+
+import mcscheduler.commons.core.Messages;
+import mcscheduler.commons.core.index.Index;
+import mcscheduler.logic.commands.RoleEditCommand;
+import mcscheduler.logic.parser.exceptions.ParseException;
+import mcscheduler.model.role.Role;
+
+/**
+ * Parses input arguments and creates a new RoleEditCommand object
+ */
+public class RoleEditCommandParser implements Parser<RoleEditCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the RoleEditCommand
+     * and returns a RoleEditCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RoleEditCommand parse(String args) throws ParseException {
+        String[] argArray = args.trim().split(" ", 2);
```

Comment:
Perhaps splitting by "\\\s+" instead of " " will be better in that users who accidentally put extra spaces won't brick and edit an existing role to a blank role. E.g. `role-edit 1         blablabla` will still work